src/Interpreters/join_common.cpp

```diff
@@ -622,9 +622,8 @@ NotJoinedBlocks::NotJoinedBlocks(std::unique_ptr<RightColumnsFiller> filler_,
         setRightIndex(right_pos, result_position);
     }
 
-    /// `result_sample_block` also may contains non unique column names
-    const auto & result_names = result_sample_block.getNames();
-    size_t unique_names_count = std::unordered_set<String>(result_names.begin(), result_names.end()).size();
+    /// `result_sample_block` may contains non unique column names
+    size_t unique_names_count = result_sample_block.getNamesToIndexesMap().size();
```

Comment:
I'm not sure why `getNamesToIndexesMap` create a copy of the map, but just to get the number of unique columns it's not necessary. I'd add a method to get the number explicitly.

Reply:
I'd add `const` to `unique_names_count` but it's already nit-picking. It's just a habit to propagate semantics that the variable is not supposed to change.